libtransmission/peer-mgr.cc

```diff
@@ -1995,23 +1792,7 @@ auto constexpr MaxUploadIdleSecs = time_t{ 60 * 5 };
 void closePeer(tr_peer* peer)
 {
     TR_ASSERT(peer != nullptr);
-    auto const* const s = peer->swarm;
-
-    /* if we transferred piece data, then they might be good peers,
-       so reset their `num_fails' weight to zero. otherwise we connected
-       to them fruitlessly, so mark it as another fail */
-    if (auto* const atom = peer->atom; atom->piece_data_time != 0)
-    {
-        tr_logAddTraceSwarm(s, fmt::format("resetting atom {} num_fails to 0", peer->display_name()));
-        atom->num_fails = 0;
-    }
-    else
-    {
-        ++atom->num_fails;
-        tr_logAddTraceSwarm(s, fmt::format("incremented atom {} num_fails to {}", peer->display_name(), atom->num_fails));
-    }
-
-    tr_logAddTraceSwarm(s, fmt::format("removing bad peer {}", peer->display_name()));
+    tr_logAddTraceSwarm(peer->swarm, fmt::format("removing bad peer {}", peer->display_name()));
```

Comment:
I think the original implementation makes more sense 🤔 

Sometimes peers are disconnected simply because the peer limit is reached.

What is your thinking here?

Reply:
Similar to #5756, that there's more to a successful peer connection than exchanging piece data, that PEX even if we're both seeds is valuable.

Setting num_fails to 0 when piece data is transferred probably still makes sense, but we shouldn't increment it when there's no piece data.

Reply:
Makes sense. I have no more comments regarding this.